### DISCUSSION TITLE
Export photos that are not imports

### DISCUSSION THREAD

**Discussion by pylchung** [RhetTbull/osxphotos#820] [General]
Hi,

Is there a way of matching photos that have not been imported into Photos?

I want to exclude photos that were imported from an external source, such as photos I imported on my DSLR. I am currently doing a regex match on {exif.camera_make} equal to "Apple", but photos imported from another iPhone user would get matched too.

Ideally I would like to match all photos that Photos that are not in the "Imports" section in Photos.

**Comment by RhetTbull** [comment]
Yes.  Every photo that has been imported has an [import_info](https://github.com/RhetTbull/osxphotos/blob/master/API_README.md#importinfo) property that includes information on when the photo was imported, what other photos were imported at the same time, etc. For any photo that was not imported, the `import_info` property will have value `None`.  Thus, you can use the `--query-eval` option to filter these:

`--query-eval "photo.import_info is None"`

The `--query-eval` option requires some understand of the internals of osxphotos (accessible through the [API_README](https://github.com/RhetTbull/osxphotos/blob/master/API_README.md)) but gives you a lot of control over filtering photos.

**Reply by pylchung** [reply]
Thanks! I'll give that a go!

**Comment by pylchung** [comment]
That works a treat. Thanks very much!
